contracts/interfaces/ICover.sol

```diff
@@ -9,27 +9,44 @@ interface ICover is IERC721 {
   /* ========== DATA STRUCTURES ========== */
 
   struct StakingPool {
-    uint160 id;
-    uint96 bookedAmount;
+    address poolAddress;
+    uint96 coverAmount;
+    uint96 premiumInNXM;
   }
 
   struct Cover {
     uint24 productId;
     uint8 payoutAsset;
-    uint8 deniedClaims;
     uint96 amount;
     uint32 start;
     uint32 period;  // seconds
+    uint96 premium;
   }
 
   struct Product {
-    uint8 payoutAsset;
+    uint16 productType;
     address productAddress;
+    uint16 capacityFactor;
+    /* supported payout assets bitmap TODO: explain */
+    uint payoutAssets;
+  }
+
+  struct ProductType {
+    string descriptionIpfsHash;
+    uint8 redeemMethod;
+    uint16 gracePeriodInDays;
+    uint16 burnRatio;
```

Comment:
if we go with the latest suggestion from Hugh, to burn proportional to the capacityFactor, then we don't need to store this.

example:
Stake of 100 NXM
Total Capacity of 200 NXM (capacity factor of 2x)
Claim of 80 NXM (example)
Stake burn of 40 NXM